.github/workflows/create_model_card_and_spec.py

```diff
@@ -0,0 +1,291 @@
+            'data_sites': {
+                'number_of_sites': model_spec.number_of_sites,
+                'sites': model_spec.sites
+            },
+            'scanner_models': model_spec.scanner_models,
+            'hardware': model_spec.hardware,
+            'training_parameters': {
+                'input_shape': model_spec.input_shape,
+                'block_shape': model_spec.block_shape,
+                'n_classes': model_spec.n_classes,
+                'lr': model_spec.lr,
+                'n_epochs': model_spec.n_epochs,
+                'total_batch_size': model_spec.total_batch_size,
+                'number_of_gpus': model_spec.number_of_gpus,
+                'loss_function': model_spec.loss_function,
+                'metrics': model_spec.metrics,
+                'data_preprocessing': model_spec.data_preprocessing,
+                'data_augmentation': model_spec.data_augmentation
+            }
+        }
+    }
+
+    with open(model_path, 'w') as file:
+        oyaml.dump(spec_data, file)
+
+def get_pull_request_description():
+    # Fetch the GitHub Pull Request event payload using GitHub API
+    pr_event_url = os.environ.get('GITHUB_EVENT_PATH', None)
+    if pr_event_url is None:
+        print("Error: GITHUB_EVENT_PATH environment variable not set.")
+        exit()
+
+    with open(pr_event_url, 'r') as file:
+        pr_event_data = oyaml.safe_load(file)
+
+    # Extract the pull request description from the payload
+    pull_request_description = pr_event_data.get(
+        'pull_request', {}).get('body', None)
+
+    if pull_request_description is None:
+        print("Error: Pull Request Description not found in the event payload.")
+        exit()
+    else:
+        return pull_request_description
+
+
+def get_dockerfile_path(model_folder):
+    for root, _, files in os.walk(model_folder):
+        for file in files:
+            if file.lower() == "dockerfile":
+                return os.path.join(root)
+    return None
+
+
+def extract_organization_name(pull_request_description):
+    try:
+        data = oyaml.safe_load(pull_request_description)
+        model_details = data.get('Model Details', {})
+        org_name = model_details.get('Organization Name', None)
+        return org_name
+    except Exception as e:
+        print(f"Error while parsing YAML: {e}")
+        return None
+
+
+def get_latest_model_name():
+    pull_request_description = get_pull_request_description()
+
+    org_folder = extract_organization_name(pull_request_description)
+
+    # Cd into the org folder
+    os.chdir(org_folder)
+
+    # Assign current directory to model_n
+    model_n = os.getcwd()
+
+    if model_n == 'objects':
+        exit()
+    else:
+        return model_n
+
+
+if __name__ == "__main__":
+    # Get the PR description
+    pr_description = oyaml.safe_load(get_pull_request_description())
+
+    # Get the latest model name
+    model_folder = get_latest_model_name()
```

Comment:
Can we not pass `pr_description` into this function to avoid calling `get_pull_request_description()` a second time inside?